python/ray/util/sgd/v2/worker_group.py

```diff
@@ -77,16 +98,32 @@ def __init__(self,
         self.num_gpus_per_worker = num_gpus_per_worker
         self.additional_resources_per_worker = additional_resources_per_worker
         self.workers = []
+        if actor_cls:
+            if not issubclass(actor_cls, BaseWorkerMixin):
+                self._base_cls = create_executable_class(actor_cls)
+            else:
+                # The provided class already subclasses the mixin.
+                self._base_cls = actor_cls
+        else:
+            self._base_cls = BaseWorkerMixin
+
+        self._actor_cls_args = actor_cls_args or []
+        self._actor_cls_kwargs = actor_cls_kwargs or {}
+
         # TODO(matt): Validate resources. Fast-fail if it is impossible to
         #  handle the request, rather than hang indefinitely.
         self._remote_cls = ray.remote(
             num_cpus=self.num_cpus_per_worker,
             num_gpus=self.num_gpus_per_worker,
-            resources=self.additional_resources_per_worker)(BaseWorker)
+            resources=self.additional_resources_per_worker)(self._base_cls)
         self.start()
 
     def _create_worker(self):
-        return self._remote_cls.remote()
+        if type(self._base_cls) == BaseWorkerMixin:
+            return self._remote_cls.remote()
+        else:
+            return self._remote_cls.remote(*self._actor_cls_args,
+                                           **self._actor_cls_kwargs)
```

Comment:
Can this just be simplified to the single line:
```
return self._remote_cls.remote(*self._actor_cls_args, **self._actor_cls_kwargs)
```

If anything I think it makes sense to move validation of  `actor_cls`, `actor_cls_args`, and `actor_cls_kwargs` to `__init__`.

Reply:
Good point, updated!